dateparser/custom_lang_detect/fasttext.py

```diff
@@ -0,0 +1,33 @@
+import fasttext
+import os
+
+from dateparser.conf import apply_settings
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+
+_model = "lid.176.ftz"
+_model_path = dir_path + '/' + _model
+
+if not os.path.exists(_model_path):
+    import urllib.request
+    print("Downloading model", _model)
+    url = "https://dl.fbaipublicfiles.com/fasttext/supervised-models/" + _model
+    try:
+        urllib.request.urlretrieve(url, _model_path)
+    except urllib.error.HTTPError:
+        print("Fasttext model", _model, "cannot be downloaded due to HTTP error")
+        raise urllib.error.HTTPError
```

Comment:
I think we wanted to have a model download command instead of downloading the model on the module level?

Reply:
I am working on it

Reply:
Hi, the implementation will require users to download the models using 
`python3 dateparser/custom_lang_detect/fasttext.py download model_name`

So would I go ahead with it or make custom settings just for the model name? 
And should there be any default model if the user does not explicitly download any model.
Pl suggest

Reply:
> ``python3 dateparser/custom_lang_detect/fasttext.py download model_name``

this can be simplified to something like

```
dateparser-download fasttext
# or for a non-default model
dateparser-download fasttext model_name
```

such scripts can be exposed in ``setup.py`` with ``entry_points``.

I think it would be great to place the model into the user cache folder, like other libraries do, for example pytorch or nltk.

Regarding whether to download the model automatically or not - I think both approaches are fine, and more "modern" libraries seem to default to downloading the model automatically. 

Examples how downloading is implemented in torch: https://github.com/pytorch/pytorch/blob/75024e228ca441290b6a1c2e564300ad507d7af6/torch/hub.py - it has a lot of stuff and we don't need all of that, but some parts are interesting, for example how they get the folder where to store the models.

Reply:
Thanks

Reply:
Hi @lopuhin, I have implemented the entry point can you please check

Reply:
Great @gavishpoddar - as discussed, let's store the model in the user cache folder, e.g. similar to above example.